src/vivarium_nih_us_cvd/components/healthcare_visits.py

```diff
@@ -171,25 +217,85 @@ def on_time_step_cleanup(self, event: Event) -> None:
 
         self.population_view.update(
-            df[[self.visit_type_column, self.scheduled_visit_date_column]]
+            df[
+                [
+                    self.visit_type_column,
+                    self.scheduled_visit_date_column,
+                    self.sbp_medication_column,
+                ]
+            ]
+        )
+
+    def calculate_medication_coverage_probabilities(self, df: pd.DataFrame) -> Dict:
+        """Determine the probability of each simulant being medicated"""
+        medication_coverage_covariates = {}
+        med_types = data_values.BASELINE_MEDICATION_COVERAGE_COEFFICIENTS._fields
+        for med_type in med_types:
+            (
+                c_int,
+                c_sbp,
+                c_ldlc,
+                c_age,
+                c_sex,
+            ) = eval(f"data_values.BASELINE_MEDICATION_COVERAGE_COEFFICIENTS.{med_type}")
+            medication_coverage_covariates[med_type] = np.exp(
+                c_int
+                + c_sbp * self.sbp(df.index)
+                + c_ldlc * self.ldlc(df.index)
+                + c_age * df["age"]
+                + c_sex * df["sex"].map(data_values.BASELINE_MEDICATION_COVERAGE_SEX_MAPPING)
+            )
+        # Calculate probabilities of being medicated
+        p_medication = {}
+        p_denominator = sum(medication_coverage_covariates.values()) + 1
+        for med_type in med_types:
+            p_medication[med_type] = pd.Series(
+                medication_coverage_covariates[med_type] / p_denominator, name=med_type
+            )
+        p_medication["NONE"] = pd.Series(1 / p_denominator, name="NONE")
+
+        return p_medication
+
+    def initialize_medication_coverage(self, df: pd.DataFrame):
+        """Initializes medication coverage"""
+        p_medication = self.calculate_medication_coverage_probabilities(df)
+        df_p_meds = pd.concat(p_medication, axis=1)
+        medicated_states = self.randomness.choice(
+            df_p_meds.index, choices=df_p_meds.columns, p=np.array(df_p_meds)
+        )
+        medicated_sbp = medicated_states[medicated_states.isin(["SBP", "BOTH"])].index
+        medicated_ldlc = medicated_states[medicated_states.isin(["LDLC", "BOTH"])].index
+
+        # Define what level of medication for the medicated simulants
+        df.loc[medicated_sbp, self.sbp_medication_column] = self.randomness.choice(
+            medicated_sbp,
+            choices=list(data_values.BASELINE_MEDICATION_LEVEL_PROBABILITY["sbp"].keys()),
+            p=list(data_values.BASELINE_MEDICATION_LEVEL_PROBABILITY["sbp"].values()),
+        )
+        df.loc[medicated_ldlc, self.ldlc_medication_column] = self.randomness.choice(
+            medicated_ldlc,
+            choices=list(data_values.BASELINE_MEDICATION_LEVEL_PROBABILITY["ldlc"].keys()),
+            p=list(data_values.BASELINE_MEDICATION_LEVEL_PROBABILITY["ldlc"].values()),
         )
 
+        return df
+
```

Comment:
You should be specific in your typing of `visitors`. I think it's `Dict[str, pd.Index]`?

That being said, I've commented above that I think it's better to set the visit type outside this function and just pass and index.

Reply:
Oops, yeah I need to be better about actually running `pytype`. But yes, irrelevant if I end up moving the visit type functionality out

Reply:
The reason for this was to deal with simulants who both miss their scheduled appt and also go for a background. We checked with Syl and she confirmed if a simulant misses a scheduled one it's fine to not allow them to go in for background. I will remove this dictionary nonsense.